modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/DefaultExecutionMediator.java

```diff
@@ -146,7 +150,11 @@ private OperationExecutionResult executeWithInterceptors(OperationExecutor execu
       after(context, result, interceptors);
     }
 
-    return new OperationExecutionResult(result, exception, Optional.ofNullable(retryRequestHolder.get()));
+    Mono<Object> reactiveResult = result instanceof Publisher
+        ? from((Publisher) result)
+        : justOrEmpty(result);
```

Comment:
In what case would empty need to be returned?  Empty will filter event in flow.. unless it's handled in `OperationMessageProcessor` specifically.